### ISSUE TITLE
/src/PhpSpreadsheet/Reader/Csv.php:280 - auto_detect_line_endings is deprecated

### ISSUE THREAD

**Issue by laurin1** [PHPOffice/PhpSpreadsheet#2609]
I see the comment that this is deprecated, but I don't see an issue open to replace/remove this code, so I'm opening one as we just moved to 8.1 and are getting this warning. We'll suppress it for now, but just wanted to make sure it was addressed:

![image](https://user-images.githubusercontent.com/75857/154847041-7f757559-3765-40b6-a003-7e1dff715bac.png)

**Comment by oleibman** [comment]
The warning message is already suppressed. Why are you seeing a warning?

**Comment by oleibman** [comment]
It will be removed when it is no longer supported. Deprecated or not, it is supported in 8.1, and removing might (agree that it is unlikely) be a problem for existing users. There is already a unit test in place which will fail when it is no longer supported, and we will respond to the test failure when that time comes.

**Comment by laurin1** [comment]
Ok, that's fine. Like I said, I did not see an issue open (or closed) for it, so I wanted to make sure it was not missed (or submit a patch if needed).

**Comment by oleibman** [comment]
Thank you for being proactive.

The test which will eventually fail is Reader/Csv/CsvLineEndingTest. I will leave this issue open for about a week to see if it sparks any other discussion.

**Comment by oleibman** [comment]
Closing. Merged PR 2623 to give people a means to avoid deprecation notice even with custom error handler. Test which will fail in Php9 should guarantee that the deprecation is not overlooked.

**Comment by Onerosolutions** [comment]
I am still facing this issue in latest PhpSpreadshet, so i decited to comment ```src\PhpSpreadsheet\Reader\Csv.php line 365 & 425``` for temporary solutions.

![image](https://github.com/PHPOffice/PhpSpreadsheet/assets/127706019/cf70ca7d-b9b5-43c7-a098-bfd1ccdc5583)

**Comment by oleibman** [comment]
There is no need to comment out any code. Add the following to your application instead.
```php
$reader = new \PhpOffice\PhpSpreadsheet\Reader\Csv();
$reader->setTestAutoDetect(false);
```

**Comment by Onerosolutions** [comment]
> There is no need to comment out any code. Add the following to your application instead.
> 
> ```
> $reader = new \PhpOffice\PhpSpreadsheet\Reader\Csv();
> $reader->setTestAutoDetect(false);
> ```

I actually tried something like that before. Because there is a setTestAutoDetect() method when I looked at the source code, but when I tried it, it says undefined.

![image](https://github.com/PHPOffice/PhpSpreadsheet/assets/127706019/e65264cc-9c8c-44f8-8067-3160788ca0c2)

It turns out the instance calling is different... So, I just need to modify the instance according to the input type, using either new \PhpOffice\PhpSpreadsheet\Reader\Csv() or Xlsx(), right?
